atomic-primops/Data/Atomics.hs

```diff
@@ -349,7 +349,21 @@ casMutVar2 mv tick new = IO$ \st ->
 -- | Memory barrier implemented by the GHC rts (see SMP.h).
 -- writeBarrier :: IO ()
 
-#if !(defined(mingw32_HOST_OS) && __GLASGOW_HASKELL__ < 802)
+#if __GLASGOW_HASKELL__ > 909
```

Comment:
Should this be `>= 909` instead of `> 909`? My impression is that GHC HEAD currently uses the version 9.9.*, so it wouldn't be able to benefit from these changes unless we use `>= 909`.